### ISSUE TITLE
feat: aux_data for inline macros

### ISSUE THREAD

**Issue by shramee** [starkware-libs/cairo#4262]
# Feature Request

**Describe the Feature Request**

Inline macros returning `PluginGeneratedFile` don't get their aux_data added to the module data.

**Describe Preferred Solution**

aux_data from inline macros be added to the module data.

**Describe Alternatives**

Alternative is to,
1. Leave a flag in code to search through the the whole tree all over again at a later point to find the parts of interest.
2. Use a static mutable var to hold a reference to areas of interest (what aux_data probably does).

**Related Code**

```rs
        InlinePluginResult {
            code: Some(PluginGeneratedFile {
                name: "set_inline_macro".into(),
                content: builder.code,
                diagnostics_mappings: builder.diagnostics_mappings,
                aux_data: Some(DynGeneratedFileAuxData::new(DojoWritersAuxData(writers))),
                // ^^^^^^ Gets ignored
            }),
            diagnostics: vec![],
        }
```

**Additional Context**

Inline macros computation: [aux_data ignored](https://github.com/starkware-libs/cairo/blob/0f0b3786e7dfdbbff681cfbb78a9dd181f2c4608/crates/cairo-lang-semantic/src/expr/compute.rs#L293)

Normal macros: [aux_data added](https://github.com/starkware-libs/cairo/blob/0f0b3786e7dfdbbff681cfbb78a9dd181f2c4608/crates/cairo-lang-defs/src/db.rs#L389)

**If the feature request is approved, would you be willing to submit a PR?**
_(Help can be provided if you need assistance submitting a PR)_

- [x] Yes
- [ ] No

**Comment by orizi** [comment]
it is not ignored - it is in the generated file.
exactly in the location the generated item **syntax** is generated you also have the aux data.

**Comment by shramee** [comment]
Thanks for getting back to me @orizi but,
1. I tried debugging with a breakpoint on `generate_code` in the macro. And it does seem to happen [here](https://github.com/starkware-libs/cairo/blob/0f0b3786e7dfdbbff681cfbb78a9dd181f2c4608/crates/cairo-lang-semantic/src/expr/compute.rs#L298). I don't see `aux_data` being utilised.
2. I'm looping through scarb main `crate_ids` looking at `module_generated_file_infos` for each module but I don't get the added `aux_data` from inline_macros, but I do for other macros.

**Comment by orizi** [comment]
oh sorry - i didn't realize you were talking about inline macros aux data, and not regular plugins.

why do you need this for inline macros?

**Comment by shramee** [comment]
#### TLDR;
To pass syntax nodes with associated module to get the semantic type.

#### LONG;
So we have some macros to read/write components (struct) on the world contract...
I need the (struct) type of `ExprPath`s passed to `set!(...)` macro, but the macro `generate_code` gets the `SyntaxDB`.
So I wanna put them together with associated module. Later we parse this to automatically give relevant permissions to writer contracts.

At the moment I'm using some static stuff,
```
type ModuleName = String;
type FunctionName = String;

lazy_static::lazy_static! {
    pub static ref WRITERS: Mutex<HashMap<ModuleName, HashMap<FunctionName, Vec<SyntaxNode>>>> = Default::default();
}
```

I think this is something aux data should be able to streamline for us...

**Comment by orizi** [comment]
I think you shouldn't use aux data from inline macros.
as you wouldn't have been able to use inline macros in rust in exactly the same way.
you CAN find the declared inline macros in your code from the regular plugin and do whatever you want to add to the aux data there.
Aux data should be in the realm of procedural macros and not inline macros.

**Comment by shramee** [comment]
I see. Yeah, That sounds fair... 👍
